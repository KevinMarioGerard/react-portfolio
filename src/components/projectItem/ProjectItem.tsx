import * as React from 'react';
import IconButton, { IconButtonProps } from '../button/IconButton';

type ProjectItemProps = {
    /**
     * Project title
     */
    title: string;

    /**
     * Project description
     */
    description: string;

    /**
     * Project tech list
     */
    techList?: Array<string>;

    /**
     * Footer icon buttons
     */
    footerButtons?: Array<IconButtonProps>;
};

const ProjectItem: React.FC<ProjectItemProps> = ({
    title,
    description,
    techList = [],
    footerButtons = [],
}) => (
    <div className="project__item">
        <h3 className="project__item__header">{title}</h3>
        <div className="project__item__description">{description}</div>
        <ul className="project__tech__list">
            {techList.map(tech => (
                <li>
                    <span>{tech}</span>
                </li>
            ))}
        </ul>
        <footer className="project__item__footer">
            {footerButtons.map(button => (
                <IconButton {...button} />
            ))}
        </footer>
    </div>
);

export default ProjectItem;
