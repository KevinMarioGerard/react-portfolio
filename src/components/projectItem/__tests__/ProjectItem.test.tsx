import { render } from '@testing-library/react';
import ProjectItem from '../ProjectItem';
import * as React from 'react';

const TestComponent = () => (
    <ProjectItem
        title="React Portfolio"
        description="My personal portfolio website I designed and built. Learned quite a bit about UI development, SEO, etc."
        techList={['ReactJS', 'Typescript', 'SCSS']}
        footerButtons={[{ icon: 'fa-github', redirectionUrl: 'mockUrl' }]}
    />
);

describe('Project Item Component tests', () => {
    it('should render without errors', () => {
        expect(() => {
            render(<TestComponent />);
        }).not.toThrow();
    });

    it('should match snapshot', () => {
        const { container } = render(<TestComponent />);

        expect(container).toMatchSnapshot();
    });
});
