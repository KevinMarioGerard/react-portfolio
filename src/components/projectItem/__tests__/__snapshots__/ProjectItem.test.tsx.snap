// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Item Component tests should match snapshot 1`] = `
<div>
  <div
    class="project__item"
  >
    <h3
      class="project__item__header"
    >
      React Portfolio
    </h3>
    <div
      class="project__item__description"
    >
      My personal portfolio website I designed and built. Learned quite a bit about UI development, SEO, etc.
    </div>
    <ul
      class="project__tech__list"
    >
      <li>
        <span>
          ReactJS
        </span>
      </li>
      <li>
        <span>
          Typescript
        </span>
      </li>
      <li>
        <span>
          SCSS
        </span>
      </li>
    </ul>
    <footer
      class="project__item__footer"
    >
      <div
        class="icon icon__container"
      >
        <i
          class="fa fa-github"
        />
      </div>
    </footer>
  </div>
</div>
`;
